resolver/resolver.cc

```diff
@@ -230,6 +230,13 @@ class ResolveConstantsWalk {
             }
             job.out->symbol = resolved;
             return;
+        } else if (resolved.exists() && resolved.data(ctx)->isClassClass()) {
+            // this can happen when we try to resolve a constant through a type alias which was itself not
+            // well-formed. If we've gotten here, we already have shown reasonable error messages to the user, so just
+            // make sure we can continue gracefully by stubbing in an untyped
+            resolved.data(ctx)->resultType = core::Types::untyped(ctx, resolved);
```

Comment:
I'm concerned of this branch being hit in other surprising cases.
Can we emit another error here?

Reply:
I also am not sure if thats the right way to handle this. 
The fact that handling the previous error has created a case that wasn't handled by other code suggests to me that the buggy code is the code that recovers from error.
Could you see if we can change that code instead?